packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -949,19 +943,31 @@ class VM extends ServiceObjectOwner {
     return invokeRpcRaw('_getVMTimeline', timeout: kLongRequestTimeout);
   }
 
-  Future<void> refreshViews() {
+  Future<void> refreshViews({ bool waitForViews = false }) async {
+    assert(waitForViews != null);
+    assert(loaded);
     if (!isFlutterEngine)
-      return Future<void>.value(null);
-    _viewCache.clear();
-    final Completer<void> completer = Completer<void>();
-    final List<Future<ServiceObject>> futures = <Future<ServiceObject>>[];
-    for (Isolate isolate in isolates.toList()) {
-      futures.add(vmService.vm.invokeRpc<ServiceObject>('_flutter.listViews',
+      return;
+    int failCount = 0;
+    while (true) {
+      _viewCache.clear();
+      final List<Future<void>> futures = <Future<void>>[];
+      for (Isolate isolate in isolates.toList()) {
+        futures.add(vmService.vm.invokeRpc<ServiceObject>(
```

Comment:
Add a comment explaining that upon completion of this RPM, `_viewCache` will be updated.

Reply:
done